File: src/main/java/top/rammer/multihighlight/ui/ChooserPanel.java
package top.rammer.multihighlight.ui;

import com.intellij.openapi.editor.markup.TextAttributes;

import org.jetbrains.annotations.NotNull;

import java.awt.event.ActionEvent;
import java.util.EventListener;

import javax.swing.JComponent;

/**
 * Created by Rammer on 07/02/2017.
 */
public interface ChooserPanel {

    @NotNull
    JComponent getPanel();

    void resetDefault();

    void reset(@NotNull TextAttributes namedTextAttr);

    void apply(@NotNull TextAttributes namedTextAttr);

    void addListener(@NotNull ColorChangedListener listener);
    
    interface ColorChangedListener extends EventListener {

        void onColorChanged(ActionEvent e);
    }
}

File: src/main/java/com/github/huoguangjin/multihighlight/ui/ChooserPanel.kt
package com.github.huoguangjin.multihighlight.ui

import com.intellij.openapi.editor.markup.TextAttributes
import java.awt.event.ActionEvent
import java.util.*
import javax.swing.JComponent

/**
 * [com.intellij.application.options.colors.OptionsPanelImpl.ColorDescriptionPanel]
 */
interface ChooserPanel {

  val panel: JComponent

  fun resetDefault()

  fun reset(textAttr: TextAttributes)

  fun apply(textAttr: TextAttributes)

  fun addListener(listener: ColorChangedListener)

  fun interface ColorChangedListener : EventListener {
    fun onColorChanged(e: ActionEvent)
  }
}
